app/concepts/application_contract.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class ApplicationContract < Dry::Validation::Contract
```

Comment:
Let add to contract instance getters with validated params keys:
```ruby
class ApplicationContract < Dry::Validation::Contract
  config.messages.default_locale = :en

  def self.call(params, &block)
    params = params.to_unsafe_hash if params.is_a?((ActionController::Parameters))
    new.call(params, &block).dup.instance_eval do
      values.each { |method, value| define_singleton_method(method) { value } }
      freeze
    end
  end
end
```

Reply:
Done